common/src/main/java/io/netty/util/ThreadDeathWatcher.java

```diff
@@ -142,22 +146,19 @@ public void run() {
                 fetchWatchees();
                 notifyWatchees();
 
-                // Try once again just in case notifyWatchees() triggered watch() or unwatch().
-                fetchWatchees();
-                notifyWatchees();
-
                 try {
                     Thread.sleep(1000);
                 } catch (InterruptedException ignore) {
                     // Ignore the interrupt; do not terminate until all tasks are run.
                 }
 
                 if (watchees.isEmpty() && pendingEntries.isEmpty()) {
+                    Thread currentThread = Thread.currentThread();
 
                     // Mark the current worker thread as stopped.
                     // The following CAS must always success and must be uncontended,
                     // because only one watcher thread should be running at the same time.
-                    boolean stopped = started.compareAndSet(true, false);
+                    boolean stopped = watcherThread.compareAndSet(currentThread, null);
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Remove this useless assignment to local variable "stopped". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://garage.netty.io/sonarqube/coding_rules#rule_key=squid%3AS1854)